torch/_inductor/codegen/common.py

```diff
@@ -320,6 +320,10 @@ def _print_GreaterThan(self, expr):
         # Go figure...
         return " >= ".join(map(self.paren, map(self._print, expr.args)))
 
+    def _print_Piecewise(self, expr):
+        # This is a hack, but we only ever emit `Piecewise` when converting bool to int
+        return f"bool({expr.args[0]}[1])"
```

Comment:
```suggestion
        return f"int({expr.args[0]}[1])"
```

Reply:
Or actually we shouldn't even need this for python code, since they're already integers.

Reply:
Oh right, we just need to emit the boolean value 😄 no wonder this worked.

Reply:
For triton you will need explicit promotion